src/relay/transforms/simplify_expr.cc

```diff
@@ -75,6 +75,59 @@ class SimplifyReshape : public DFPatternRewrite {
   DFPattern x_;
 };
 
+/*!
+ * \brief SimplifyCastLike matches the pattern of cast data to the same dtype.
+ */
+class SimplifyCastLike : public DFPatternRewrite {
+ public:
+  SimplifyCastLike() {
+    data_pat_ = IsWildcard();
+    like_pat_ = IsWildcard();
+    pattern_ = IsOp("cast_like")({data_pat_, like_pat_});
+  }
+
+  Expr Callback(const Expr& pre, const Expr& post,
+                const Map<DFPattern, Array<Expr>>& node_map) const override {
+    auto data = node_map[data_pat_][0];
+    const TensorTypeNode* data_ty = data->checked_type().as<TensorTypeNode>();
+    const TensorTypeNode* like_ty = pre->checked_type().as<TensorTypeNode>();
+    if (like_ty->dtype == data_ty->dtype) {
+      return data;
+    }
+    return post;
+  }
+
+ protected:
+  DFPattern data_pat_;
+  DFPattern like_pat_;
+};
+
+/*!
+ * \brief SimplifyCast matches the pattern of cast data to the same dtype.
+ */
+class SimplifyCast : public DFPatternRewrite {
```

Comment:
Seems like these two patterns be merged to one?
```
pattern_ = IsOp("cast_like")({data_pat_, like_pat_}) || IsOp("cast")({data_pat_});
```
And both patterns can use `like_ty = pre->checked_type().as<TensorTypeNode>();`.